### DISCUSSION TITLE
Incoming call and Audio

### DISCUSSION THREAD

**Discussion by dvxgit-tbautista** [flutterjanus/flutter_janus_client#140] [Q&A] [answered]
Hello!

I was editing and testing some stuff and I commented out the part when it hangs up, it disposes the session and sip. I wanna keep the sip and session alive then continue on making incoming calls.

I am encountering a problem where after receiving and accepting the incoming calls after the first one, the audio doesn't work anymore. Any idea what the problem could be?

Thank you very much in advance and I would appreciate your help!

**Comment by shivanshtalwar** [comment]
I need more code context to fully understand your problem, can you please share a reproducible code snippet?

**Reply by dvxgit-tbautista** [reply]
[4229593070285629] Creating ICE agent (ICE Full mode, controlling)
[4229593070285629] The DTLS handshake has been completed
[janus.plugin.sip-0x7fea2000a600] WebRTC media is now available
[janus.plugin.sip-0x7fea2000a600] No WebRTC media anymore
[4229593070285629] WebRTC resources freed; 0x7fea500022c0 0x7fea50007ba0
[4229593070285629] Creating ICE agent (ICE Full mode, controlling)
[janus.plugin.sip-0x7fea2000a600] No WebRTC media anymore
[4229593070285629] WebRTC resources freed; 0x7fea500022c0 0x7fea50007ba0

Hi, the snippet above is the trace log in the Janus Gateway.

The first 5 lines are the logs for the successful incoming call. Then after that, the second incoming call works but the audio is not working.

The code is pretty much the same with the repo's.

**Reply by dvxgit-tbautista** [reply]
Hello, should I email you a snippet of the code? Please, thank you!

**Reply by shivanshtalwar** [reply]
Have you deployed your own janus? If so which environment you are on? I would advise you to use docker project that we created specifically to avoid such problems locally as well as on production
https://github.com/flutterjanus/JanusDocker
Also i am assuming you are using some third party sip server provider and not your own deployment of freeswitch or asterisk coz configuration can be tricky to set up. Let me know what's the status

**Reply by dvxgit-tbautista** [reply]
Hello! Thanks for the reply. We set up our own Janus. 

Oracle Linux - 8.7
Janus Gateway - Latest 
Asterisk - 16

**Reply by dvxgit-tbautista** [reply]
For some unknown reason, the "await localMediaSetup();" does not work the 2nd time for incoming call so there's no audio.

**Reply by shivanshtalwar** [reply, marked as answer]
I try calling `await sip?.initializeWebRTCStack();` before initiating incoming call

**Reply by dvxgit-tbautista** [reply]
Should I include that inside the Accept button's onPressed?

**Reply by dvxgit-tbautista** [reply]
Thank you so much! It is now working! 

I included it inside the if else statement for SipIncomingCallEvent.

**Reply by shivanshtalwar** [reply]
Consider supporting the project if it helped you, btw quick survey would you like to have a platform that would provide you auto scalable janus sip plugin ? Would you like to use it instead of managing and scaling up your own infrastructure?

**Reply by dvxgit-tbautista** [reply]
Hello again! I've just spoken with one of our senior developers, and they prefer the latter.

I will now bring up our support for the project with the entire team and will get back to you!
